### PR TITLE
Validate EXAMPLES as YAML

### PR DESCRIPTION
**Pull request by sivel**
##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
Various

##### ANSIBLE VERSION

```
devel
```

##### SUMMARY
This PR validates `EXAMPLES` as YAML.  There were a number of simple mistakes, that when corrected, allow people to use the examples directly.

### CODE REVIEW

**Comment by bcoca** [comment]
funny thing, examples are not treated as YAML in docs, mostly to allow 'free form' text in there.

**Comment by sivel** [comment]
Yep, I'm aware. Although we did make a big push to update docs to use YAML formatting instead of key=value.

As it happens, the vast majority were YAML.  This effort is partially to outline that.

Should we decide to not move forward, I'm fine with that, but in the efforts of standardization, this could be beneficial.

**Comment by Fale** [comment]
I think this is a great improvement to the examples and I think should be merged :)

File: lib/ansible/modules/windows/win_path.py [hunk 1]
```diff
@@ -75,11 +75,11 @@
 - name: Ensure that system32 and Powershell are present on the global system path, and in the specified order
   win_path:
     elements:
-    - %SystemRoot%\system32
-    - %SystemRoot%\system32\WindowsPowerShell\v1.0
+    - '%SystemRoot%\system32'
+    - '%SystemRoot%\system32\WindowsPowerShell\v1.0'
```
Comment (abadger) [inline lib/ansible/modules/windows/win_path.py:79]: Did you make sure that this one generates a sane string at the end?  YAML quotes and backslashes have a lot of little wrinkles.  (Single quotes are usually the right way to do it but there are some differences between single quotes and bare strings.  I don't know if any will be hit here).

**Comment by abadger** [comment]
In general, I think this is a good thing.  Just last month, gundalow and I had to clean up a lot of things very similar to this so that warnings from sphinx-build wouldn't be a wall of text that no one would read through to pick out bigger problems.

File: lib/ansible/modules/windows/win_path.py (hunk 1 above)
Comment (sivel) [reply lib/ansible/modules/windows/win_path.py:79]: Yes, it does generate a sane string:

```
ok: [localhost] => (item=%SystemRoot%\system32) => {
    "item": "%SystemRoot%\\system32"
}
```

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Excellent, thanks for updating the tools as well as fixing the underlying issues.

FYI we now have some improved details on how to document modules. I was planning on making validate-modules reference this page when it hits an issue http://docs.ansible.com/ansible/dev_guide/developing_modules_documenting.html

Though that will be done after I've documented `ANSIBLE_METADATA`

**Comment by gundalow** [comment]
@sivel If you can rebase to resolve `lib/ansible/modules/network/bigswitch/bigmon_chain.py` I'll merge this.

**Comment by gundalow** [comment]
Thank you again for taking the time to fix these issues

**Comment by sivel** [comment]
@gundalow I've rebased, resolved conflicts, and applied YAML fixes to new modules and changes to YAML for existing modules.

Of note here, since I didn't mention it, and there aren't doc updates with this PR.  The YAML parsing of `EXAMPLES` is slightly more lenient, and supports a wider variety of info in EXAMPLES than how our other YAML documentation is parsed.

This is done, by using `safe_load_all` instead of `safe_load`, so `EXAMPLES` *can* include multiple YAML documents.  This is to support multiple structures simultaneously in a single YAML string.

An example of this would be where a full playbook is provided, and potentially just a tasks "file".  Or where we document "vars" for network modules, and also tasks.

CI is running now.

**Comment by Fale** [comment]
I think a new rebase is needed to fix a conflict in lib/ansible/modules/web_infrastructure/ansible_tower/tower_organization.py

**Comment by sivel** [comment]
I'll look soon.  Apparently some people are busy merging things today.

**Comment by Fale** [comment]
I guess the Ansible 2.3 merge window is closing down soon

**Comment by sivel** [comment]
Ok, rebased again.  CI is running.

**Comment by abadger** [comment]
gundalow gave shipit approval so I've merged to devel.

**Pull request merged**